[docs] Add draft AI tutorial (#46779)
expo/expo: docs/pages/tutorial/build-with-ai/set-up-your-tools.mdx

---
title: Set up your tools
description: In this chapter, install an AI coding agent, Node.js, and Expo Go, and teach your agent about Expo.
---

import { BookOpen02Icon } from '@expo/styleguide-icons/outline/BookOpen02Icon';

import { BoxLink } from '~/ui/components/BoxLink';
import { Collapsible } from '~/ui/components/Collapsible';
import { Prerequisites, Requirement } from '~/ui/components/Prerequisites';
import { ProgressTracker } from '~/ui/components/ProgressTracker';
import { Terminal } from '~/ui/components/Snippet';
import { Step } from '~/ui/components/Step';
import { Tabs, Tab } from '~/ui/components/Tabs';

> In this chapter, you'll install everything you need, starting from zero. This is the longest chapter — once your tools are set up, the fun starts and the rest of the tutorial moves quickly.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
In this chapter, you'll install everything you need, starting from zero. This is the longest chapter. Once your tools are set up, the fun starts and the rest of the tutorial moves quickly.
```

<Prerequisites>
  <Requirement title="A computer">A computer running macOS, Windows, or Linux.</Requirement>
  <Requirement title="A phone">
    An Android or iOS phone, connected to the same Wi-Fi network as your computer.
  </Requirement>
  <Requirement title="About 20 minutes">That's all the setup should take.</Requirement>
</Prerequisites>

<Step label="1">

## Open a terminal

> The terminal is the app where you'll talk to your AI agent. On macOS, open the built-in **Terminal** app (find it with Spotlight search). On Windows, open **PowerShell** from the Start menu. You'll type a handful of commands into it during this chapter — each one is provided for you to copy and paste.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
The terminal is the app where you'll talk to your AI agent. On macOS, open the built-in **Terminal** app (find it with Spotlight search). On Windows, open **PowerShell** from the Start menu. You'll type a handful of commands into it during this chapter. Each one is provided for you to copy and paste.
```

> If you plan to use **Cursor**, you'll work inside its visual editor instead of a terminal for most of this tutorial, but keep a terminal handy: you'll still use it to run your app.

</Step>

<Step label="2">

## Install Node.js

Node.js is the runtime that powers Expo's developer tools. Download the **LTS** version from [nodejs.org](https://nodejs.org/en) and run the installer, accepting the default options.

To confirm it worked, run the following command in your terminal. It should print a version number:

<Terminal cmd={['$ node --version']} />

</Step>

<Step label="3">

## Install an AI agent

> If you already use an AI coding agent, skip to the next step. Otherwise, pick one below — this tutorial works the same with any of them.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
If you already use an AI coding agent, skip to the next step. Otherwise, pick one below. This tutorial works the same with any of them.
```

<Tabs>

<Tab label="Claude Code">

Install [Claude Code](https://claude.com/claude-code) with the following command:

<Terminal cmd={['$ npm install -g @anthropic-ai/claude-code']} />

Then run `claude` in your terminal and follow the instructions to sign in. See the [Claude Code setup guide](https://code.claude.com/docs/en/setup) if you run into trouble.

</Tab>

<Tab label="Codex">

Install [Codex](https://developers.openai.com/codex/cli) with the following command:

<Terminal cmd={['$ npm install -g @openai/codex']} />

Then run `codex` in your terminal and follow the instructions to sign in with your ChatGPT account.

</Tab>

<Tab label="Cursor">

Download [Cursor](https://cursor.com) and install it like any other app. Cursor is a visual code editor with a built-in AI agent: you'll type prompts into its agent panel instead of a terminal.

</Tab>

</Tabs>

Other agents work too. As long as your agent can edit files and run commands on your computer, you can follow along.

</Step>

<Step label="4">

## Install Expo Go and create an Expo account

> Expo Go is a free app that runs your project on your phone while you build it — no app store publishing required. Every time your agent changes the code, the app on your phone updates within seconds.

Reviewer: **[nit]** The glossary prefers "test in Expo Go" over "run in Expo Go" phrasing, and the em dash should be restructured (`style-guide.md#glossary-and-product-names`, `style-guide.md#punctuation`).

```suggestion
Expo Go is a free app that lets you test your project on your phone while you build it, with no app store publishing required. Every time your agent changes the code, the app on your phone updates within seconds.
```

> 1. Install [Expo Go](https://expo.dev/go) from the Play Store or App Store on your phone.

Reviewer: **[nit]** Spell out the store names as proper nouns: "Google Play Store" and "Apple App Store" (`style-guide.md#glossary-and-product-names`).

```suggestion
1. Install [Expo Go](https://expo.dev/go) from the Google Play Store or Apple App Store on your phone.
```

2. Create a free account at [expo.dev/signup](https://expo.dev/signup) on your computer.
3. Open Expo Go on your phone and sign in with the same account.

</Step>

<Step label="5">

## Teach your agent about Expo

> Expo Skills are instruction files that teach AI agents how to build Expo apps well — which libraries to use, how to structure screens, and how to avoid common mistakes. Installing them is the single biggest thing you can do to get good results from your agent.

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
Expo Skills are instruction files that teach AI agents how to build Expo apps well: which libraries to use, how to structure screens, and how to avoid common mistakes. Installing them is the single biggest thing you can do to get good results from your agent.
```

<Tabs>

<Tab label="Claude Code">

Run the following command in your terminal:

<Terminal cmd={['$ claude mcp add --transport http expo https://mcp.expo.dev/mcp']} />

Then start `claude` and run `/mcp` inside it to sign in with the Expo account you created in the previous step.

</Tab>

<Tab label="Codex">

Run the following command in your terminal, and sign in with the Expo account you created in the previous step when prompted:

<Terminal cmd={['$ codex mcp add expo --url https://mcp.expo.dev/mcp']} />

</Tab>

<Tab label="Cursor">

Click the following link to install the MCP server for Cursor, then sign in with the Expo account you created in the previous step:

<a href="cursor://anysphere.cursor-deeplink/mcp/install?name=expo&config=eyJ1cmwiOiJodHRwczovL21jcC5leHBvLmRldi9tY3AifQ%3D%3D">
  <img
    src="https://cursor.com/deeplink/mcp-install-light.svg"
    alt="Install MCP Server"
    className="hidden max-w-48 dark:block"
  />
  <img
    src="https://cursor.com/deeplink/mcp-install-dark.svg"
    alt="Install MCP Server"
    className="block max-w-48 dark:hidden"
  />
</a>

</Tab>

</Tabs>

<Collapsible summary="Optional: let your agent see and tap your app">

> The MCP server also offers local capabilities: with extra setup, a multimodal agent can take screenshots of your app running in a simulator, tap buttons, and verify its own work. This requires a simulator on your computer (macOS only for iOS), so it's beyond this tutorial — in the chapters ahead, **you** are the one verifying the app on your phone. If you want to explore it later, see [Set up local capabilities](/eas/ai/mcp/#set-up-local-capabilities-recommended).

Reviewer: **[nit]** Prefer separate sentences over em dashes (`style-guide.md#punctuation`).

```suggestion
The MCP server also offers local capabilities: with extra setup, a multimodal agent can take screenshots of your app running in a simulator, tap buttons, and verify its own work. This requires a simulator on your computer (macOS only for iOS), so it's beyond this tutorial. In the chapters ahead, **you** are the one verifying the app on your phone. If you want to explore it later, see [Set up local capabilities](/eas/ai/mcp/#set-up-local-capabilities-recommended).
```
